Realm/RLMObject.mm

```diff
@@ -219,3 +219,31 @@ - (void)setValue:(id)value forUndefinedKey:(NSString *)key {
 }
 
 @end
+
+@implementation RLMWeakObjectHandle {
+    realm::Row _row;
+    RLMRealm *_realm;
+    RLMObjectSchema *_objectSchema;
+    Class _objectClass;
+}
+
+- (instancetype)initWithObject:(RLMObjectBase *)object {
+    if (!(self = [super init])) {
+        return nil;
+    }
+
+    _row = object->_row;
+    _realm = object->_realm;
+    _objectSchema = object->_objectSchema;
+    _objectClass = object.class;
```

Comment:
This doesn't actually need the object class; it can just create an RLMObjectBase. It also doesn't need the object schema, but that's a bit iffier since that relies on RLMDynamicGet() not needing it.